src/coreclr/jit/lowerxarch.cpp

```diff
@@ -8638,6 +8649,78 @@ void Lowering::ContainCheckCast(GenTreeCast* node)
 #endif // !defined(TARGET_64BIT)
 }
 
+//------------------------------------------------------------------------
+// TryLowerAndOrToCCMP : Lower AND/OR of two conditions into test + CCMP + SETCC nodes.
+//
+// Arguments:
+//    tree - pointer to the node
+//    next - [out] Next node to lower if this function returns true
+//
+// Return Value:
+//    false if no changes were made
+//
+bool Lowering::TryLowerAndOrToCCMP(GenTreeOp* tree, GenTree** next)
+{
+    assert(tree->OperIs(GT_OR, GT_AND));
+
+    if (!comp->opts.OptimizationEnabled())
+    {
+        return false;
+    }
+
+    GenTree* lastNode = BlockRange().LastNode();
+
+    if (!lastNode->OperIs(GT_JTRUE, GT_RETURN))
+    {
+        return false;
+    }
```

Comment:
This does not make sense to me. This transformation should be beneficial to do regardless of how the and/or is used.